### PR TITLE
Fixes #1109 - Documentation explaining how to use Python package from plugin.

### LINKED ISSUE #1109: Errbot Import error on the class same package with module
### I am.

* [ ] Reporting a bug
* [ ] Suggesting a new feature
* [ ] Requesting help with running my bot
* [x ] Requesting help writing plugins
* [ ] Here about something else

### I am running...

* Errbot version:5.1.2
* OS version: Debian 8
* Python version: 3.6
* Using a virtual environment: yes

### Issue description
```
|----/opt/errbot/plugins/Vm/
                         |---vm.py
                         |---vm.plug
                         |---model.py
                         |---*
```

I have a plugin module, in the plugin module directory i included a python class name M_Vm to act as a modeler in model.py

How ever, when i run errbot, im not able to import the M_Vm class

```
errbot_1  |   File "/usr/local/lib/python3.6/site-packages/yapsy/PluginManager.py", line 488, in loadPlugins
errbot_1  |     candidate_module = imp.load_module(plugin_module_name,plugin_file,candidate_filepath+".py",("py","r",imp.PY_SOURCE))
errbot_1  |   File "/usr/local/lib/python3.6/imp.py", line 234, in load_module
errbot_1  |     return load_source(name, filename, file)
errbot_1  |   File "/usr/local/lib/python3.6/imp.py", line 172, in load_source
errbot_1  |     module = _load(spec)
errbot_1  |   File "<frozen importlib._bootstrap>", line 675, in _load
errbot_1  |   File "<frozen importlib._bootstrap>", line 655, in _load_unlocked
errbot_1  |   File "<frozen importlib._bootstrap_external>", line 678, in exec_module
errbot_1  |   File "<frozen importlib._bootstrap>", line 205, in _call_with_frames_removed
errbot_1  |   File "/opt/errbot/plugins/Vm/vm.py", line 3, in <module>
errbot_1  |     from model import M_Vm
```

running pylint on vm.py doesnt complaint anything on no name module error...

```
No config file found, using default configuration
************* Module vm
C:212, 0: Trailing newlines (trailing-newlines)
C:  1, 0: Missing module docstring (missing-docstring)
R: 33, 0: Too many ancestors (11/7) (too-many-ancestors)
W: 45,33: Unused argument 'args' (unused-argument)
R: 45, 4: Method could be a function (no-self-use)
R: 51, 4: Method could be a function (no-self-use)
R: 60, 4: Method could be a function (no-self-use)
R: 70, 4: Method could be a function (no-self-use)
R:110, 4: Method could be a function (no-self-use)
W:128, 8: Redefining built-in 'list' (redefined-builtin)
R:121, 4: Method could be a function (no-self-use)
W:139,41: Unused argument 'polled_time' (unused-argument)
R:153, 4: Method could be a function (no-self-use)
E:169, 4: method already defined line 121 (function-redefined)
R:169, 4: Method could be a function (no-self-use)
C:196, 4: Missing method docstring (missing-docstring)
R:196, 4: Method could be a function (no-self-use)
C:200, 4: Missing method docstring (missing-docstring)
R:200, 4: Method could be a function (no-self-use)
C:204, 4: Missing method docstring (missing-docstring)
R:204, 4: Method could be a function (no-self-use)
C:  9, 0: Imports from package errbot are not grouped (ungrouped-imports)

```

### PR DESCRIPTION
**Pull request by nzlosh**

### CODE REVIEW

**Comment by sijis** [comment]
Would it make sense to include some snippet on how to use this `lib/` in a plugin?

**Comment by nzlosh** [comment]
Yes, it would help people better understand how the final solution should look.  I've based the example on the previous plugin example so there is a flow from the basic plugin to a more advanced plugin using submodules.

File: docs/user_guide/plugin_development/basics.rst [hunk 1]
```diff
@@ -160,6 +160,43 @@ The `[Documentation]` section will be explained in more detail
...
+    ├── README.md
+    ├── requirements.txt
+    ├── lib
+    │   ├── __init__.py
+    │   ├── moduleA.py
+    │   ├── moduleB.py
+    │   ├── moduleC.py
+
+The presence of `__init__.py` indicates `lib` is a Python regular package.  Assuming `moduleA` has
+the function `InvertString`, the `helloworld` plugin can import it and use it with the following syntax:
```
Comment (sijis) [inline docs/user_guide/plugin_development/basics.rst]: Make the function look like a function, not a class.

```suggestion
the function `invert_string()`, the `helloworld` plugin can import it and use it with the following syntax:
```

File: docs/user_guide/plugin_development/basics.rst [hunk 2]
```diff
@@ -160,6 +160,43 @@ The `[Documentation]` section will be explained in more detail
...
+
+.. code-block:: python
+
+    from lib.moduleA import InvertString
```
Comment (sijis) [inline docs/user_guide/plugin_development/basics.rst]:
```suggestion
    from lib.moduleA import invert_string
```

File: docs/user_guide/plugin_development/basics.rst [hunk 3]
```diff
@@ -160,6 +160,43 @@ The `[Documentation]` section will be explained in more detail
...
+    from errbot import BotPlugin, botcmd
+
+    class HelloWorld(BotPlugin):
+        """Example 'Hello, world!' plugin for Errbot"""
+
+        @botcmd
+        def hello(self, msg, args):
+            """Say hello to the world"""
+            return InvertString("Hello, world!")
```
Comment (sijis) [inline docs/user_guide/plugin_development/basics.rst]:
```suggestion
            return invert_string("Hello, world!")
```

**Pull request merged**